arkane/statmech.py

```diff
@@ -273,6 +276,7 @@ def create_log(log_path, check_for_errors=True):
             'True': True,
             'False': False,
             'HinderedRotor': hinderedRotor,
+            'HinderedRotor1DArray': hinderedRotor1DArray,
```

Comment:
Should we add this option to Arkane's documentation?

Reply:
Thank you again for pointing me to the documentation. I just realize `HinderedRotor` may take a text file with angles and energies stored. It turns out I never find this before...So it actually does something very similar to my addition. Do you know it is still worth adding the class `HinderedRotor1DArray`?

Reply:
The current option only allows to take in a list of angles and energies. Your option has more arguments. Perhaps keep the current option of passing a file to `ScanLog`. We can make this file more machine-friendly by allowing one to pass a yaml file, where you can add all your args and a list of data points.

Reply:
As suggested, I added the support of reading csv files and yaml files using `ScanLog`. For both method, I added unit tests and related descriptions in the documentation.